src/abstraction/transient.rs

```diff
@@ -171,6 +223,13 @@ impl TransientObjectContext {
         Ok(key)
     }
 
+    /// Sign a digest with an existing key.
+    ///
+    /// Returns the signature.
+    ///
+    /// # Errors
+    /// * errors are returned if any method calls return an error: `Context::context_load`,
+    /// `Context::sign`, `Context::flush_context`, `TransientObjectContext::set_session_attrs`
     pub fn sign(
```

Comment:
This method along with others around here make use of constraints that apply inside the methods it calls (e.g. in this case, the length of `key_auth` is checked in `set_handle_auth` so there's no need to check it here) - should we document this specifying the same constraints?

Reply:
I think this is fine not to document it in the upper layer as well as you already say that "errors are returned if any method calls return an error".

On that note, maybe nice to add `set_handle_auth` to the list then?